src/applications/personalization/profile/tests/components/contact-information/ContactInformation.delete-telephone.unit.spec.jsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { expect } from 'chai';
+import { setupServer } from 'msw/node';
+
+import { FIELD_TITLES, FIELD_NAMES } from '@@vap-svc/constants';
+
+import * as mocks from '@@profile/msw-mocks';
+import ContactInformation from '@@profile/components/contact-information/ContactInformation';
+
+import {
+  createBasicInitialState,
+  renderWithProfileReducers,
+  wait,
+} from '../../unit-test-helpers';
+
+const ui = (
+  <MemoryRouter>
+    <ContactInformation />
+  </MemoryRouter>
+);
+let view;
+let server;
+
+function getEditButton(numberName) {
+  let editButton = view.queryByText(new RegExp(`add.*${numberName}`, 'i'), {
+    selector: 'button',
+  });
+  if (!editButton) {
+    // Need to use `queryByRole` since the visible label is simply `Edit`, but
+    // the aria-label is more descriptive
+    editButton = view.queryByRole('button', {
+      name: new RegExp(`edit.*${numberName}`, 'i'),
+    });
+  }
+  return editButton;
+}
+
+function deletePhoneNumber(numberName) {
+  // delete
+  view
+    .getByLabelText(new RegExp(`remove ${numberName}`, 'i'), {
+      selector: 'button',
+    })
+    .click();
+  const confirmDeleteButton = view.getByText('Yes, remove my information', {
+    selector: 'button',
+  });
+  confirmDeleteButton.click();
+
+  return { confirmDeleteButton };
+}
+
+async function testSuccess(numberName) {
+  server.use(...mocks.transactionPending);
+
+  deletePhoneNumber(numberName);
+
+  // check that the "we're deleting your..." message appears
+  const deletingMessage = await view.findByText(
+    new RegExp(
+      `We’re in the process of deleting your ${numberName}. We’ll remove this information soon.`,
+      'i',
+    ),
+  );
+  expect(deletingMessage).to.exist;
+
+  server.use(...mocks.transactionSucceeded);
+
+  await wait(1500);
```

Comment:
Same here, anyway to reduce wait time for this unit test?  Starting to feel more like an integration test here 🤔

Reply:
Yea you are right. 

I think there is a way to set the interval / timeout for the 'transaction' status request to a lower time for unit tests specifically. I'll look into how this was done in the past. This really is an integration test at heart due to it testing the whole lifecycle of the delete flow.

This is related to the transactional style of certain api requests where they will make a PUT request and then loop on an interval and hit a status endpoint to check the status of the transaction. I would love to remove this loop at some point in the future and move it to the BE but not sure how to go about that.

Reply:
Updated